gen_conf: Use UUID for non-name-ref parent

cathay4t opened the pull request:
If parent interface is not referenced by interface name, in `gen_conf`
mode, nmstate cannot tell the real interface name of parent, so child
should refer its parent by UUID.

Implementation detail:

```
   In `gen_conf` mode, the UUID of certain interface can be predict by
   its name and type, so instead of modify the generated NmConnection
   afterwards, this patch invoke `Interface.change_parent_name()` to
   change the parent name into `UUID` of its parent when its parent is
   non-name identifier.
```

Integration test cases included for all child interface types:
* VLAN
* VxLAN
* MAC VTAP
* MAC VLAN
* MAC Sec

Resolves: https://issues.redhat.com/browse/RHEL-94783

cathay4t committed:
gen_conf: Add support of MacSec interface

Integration test case included.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
gen_conf: Support PCI address matching

Integration test case included.

Skipped in CI due to lack of hardware, manually tested.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
Fix cargo warning in unit tests

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
test: clean up after `test_vlan_add_multiple_qos_map_in_mixed_order`

Remove the test VLAN after test case
`test_vlan_add_multiple_qos_map_in_mixed_order` finished.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
Github CI: Use ubuntu-latest

Signed-off-by: Gris Ge <fge@redhat.com>

ihuguet commented on tests/integration/nm/gen_conf_test.py:
```diff
@@ -588,3 +588,181 @@ def test_gen_conf_ovs_bridge_port_ref_by_mac(eth1_eth2_up_with_no_config):
+def test_gen_conf_macsec_parent_ref_by_mac(eth1_eth2_up_with_no_config):
+    port1_mac = get_mac_address("eth1")
+
+    desired_state = load_yaml(
+        f"""---
+interfaces:
+- name: port1
+  type: ethernet
+  identifier: mac-address
+  mac-address: {port1_mac}
```
Please indent these lines

cathay4t replied:
Fixed. Indented and shorten the `mka-ckn`.

cathay4t committed:
gen_conf: Use UUID for non-name-ref parent

If parent interface is not referenced by interface name, in `gen_conf`
mode, nmstate cannot tell the real interface name of parent, so child
should refer its parent by UUID.

Implementation detail:
    In `gen_conf` mode, the UUID of certain interface can be predict by
    its name and type, so instead of modify the generated NmConnection
    afterwards, this patch invoke `Interface.change_parent_name()` to
    change the parent name into `UUID` of its parent when its parent is
    non-name identifier.

Integration test cases included for all child interface types matched by
MAC address:
 * VLAN
 * VxLAN
 * MAC VTAP
 * MAC VLAN
 * MAC Sec

Resolves: https://issues.redhat.com/browse/RHEL-94783

Signed-off-by: Gris Ge <fge@redhat.com>
